.tekton/on-push.yaml

```diff
@@ -71,9 +71,88 @@ spec:
             value: $(params.repo_url)
           - name: REVISION
             value: $(params.revision)
-      - name: buildah-pvc
+      - name: lint-and-test
         runAfter:
           - fetch-repository
+        workspaces:
+          - name: source
+            workspace: source
+        taskSpec:
+          workspaces:
+            - name: source
+          volumes:
+            - name: env-file-volume
+              secret:
+                secretName: integration-tests-env-file
+            - name: examples-volume
+              configMap:
+                name: integration-tests-reports
+          steps:
+            - name: lint-test
+              env:
+                - name: PIP_ROOT_USER_ACTION
+                  value: "ignore"
+                - name: AGENT_MORPHEUS_TRACING_ENABLED
+                  value: "false"
+              image: docker.io/condaforge/miniforge3:latest
+              workingDir: $(workspaces.source.path)
+              volumeMounts:
+                - name: env-file-volume
+                  mountPath: $(workspaces.source.path)/.env
+                  subPath: .env
+                - name: examples-volume
+                  mountPath: $(workspaces.source.path)/examples
+              script: |
+                #!/bin/bash
+                
+                set -eo pipefail 
+
+                ENV_NAME="test-env"
+                CONDA_PKGS="make pylint pytest pytest-asyncio pytest-dotenv python-dotenv openpyxl ragas seaborn"
+                PIP_PKGS="langchain_nvidia_ai_endpoints==0.2.0"
+
+                print_banner() {
+                    echo
+                    echo "----------- ${1} -----------"
+                }
+
+                print_banner "CHECKING FOR .ENV FILE"
+                if [ -f .env ]; then
+                    echo ".env file successfully mounted."
+                else
+                    echo "ERROR: .env file not found!"
+                    exit 1
+                fi
+
+                print_banner "CHECKING FOR EXAMPLES DIR"
+                if [ -d "examples" ] && [ -n "$(ls -A examples)" ]; then
+                    echo "Examples directory successfully mounted."
+                else
+                    echo "ERROR: Examples directory not found or is empty!"
+                    exit 1
+                fi
+
+                print_banner "CREATING TEST ENV"
+                mamba env create --log-level error -q -y -f requirements.yaml -n "${ENV_NAME}"
+
+                print_banner "ACTIVATING CONDA ENV"
+                eval "$(mamba shell hook -s bash)"
+                mamba activate "${ENV_NAME}"
+
+                print_banner "INSTALLING TEST DEPENDENCIES"
+                mamba install --log-level error -q -y -c conda-forge ${CONDA_PKGS}
+                pip install -q ${PIP_PKGS}
+
+                print_banner "RUNNING LINTER"
+                make lint-pr
```

Comment:
@vbelouso  Because it's on push, then it will not lint anything.
because there is no diff between HEAD ( which is rh-main on push event) and rh-main...
anyway it's not critical as it happens already on PR open/sync.....
If you insist on running the linter on delta files again , you can use the `git show HEAD` variant instead an introduce a new target on the make file such as `lint-pr-on-push`, and use the following command ( for instance, or anything equivalent):
```shell
git show HEAD --name-only --oneline -- *.py | tee head.out | tail -n $(($(cat head.out  | wc -l)-1))
```